lib/mox/server.ex

```diff
@@ -67,20 +73,28 @@ defmodule Mox.Server do
     {:reply, pending, state}
   end
 
+  def handle_call({:allow, mock, owner_pid, pid}, _from, state) do
```

Comment:
This function needs to handle two error cases:

1. The processing being allowed (pid) is already allowed by another process. This needs to return `{:error, :already_allowed}` which needs to become a `raise` back in the Mox API

2. The processing being allowed (pid) already has an expectation set on its name. This needs to return `{:error, :currently_allowed}` which also needs to become a `raise` back in the Mox API